arbnode/transaction_streamer.go

```diff
@@ -1565,6 +1565,20 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context) er
 			if err != nil {
 				return nil, err
 			}
+			if pos > 1 {
+				prevMsg, err := s.GetMessage(pos - 1)
+				if err != nil {
+					return nil, err
+				}
+				if prevMsg.DelayedMessagesRead+1 == msg.DelayedMessagesRead {
+					// This message is a delayed message, and it should not be included
+					// in the hotshot payload. The caff node is supposed to fetch the delayed message
```

Comment:
If my understanding is correct, delayed messages are included with all the messages that go through the transaction streamer. So in that case we just skip it here?

Reply:
For the batch poster, the source of its messages is the feed from sequencer. Delayed sequencer is responsible for relaying the delayed messages when it discover them in the delayed inbox, which is on the L1.

In batch poster's logic, when delayed message is found, it will add a constant segment to the building batch, not the content of the delayed message. This means other parties can just fetch the delayed messages by themselves through L1.

Our caff nodes also read delayed messages on their own. So we don't need to include the content of delayed messages in hotshot payload. Delayed sequencer could be wrong and malicious. By doing this it couldn't cheat on the messages